src/commands/CommandHandler.ts

```diff
@@ -38,90 +40,93 @@ import { execShutdownRoomCommand } from "./ShutdownRoomCommand";
-        } else if (parts[1] === 'rooms' && parts.length > 3 && parts[2] === 'remove') {
+        } else if (cmd === 'rooms' && parts.length > 3 && parts[2] === 'remove') {
             return await execRemoveProtectedRoom(roomId, event, mjolnir, parts);
-        } else if (parts[1] === 'rooms' && parts.length === 2) {
+        } else if (cmd === 'rooms' && parts.length === 2) {
             return await execListProtectedRooms(roomId, event, mjolnir);
-        } else if (parts[1] === 'move' && parts.length > 3) {
+        } else if (cmd === 'move' && parts.length > 3) {
             return await execMoveAliasCommand(roomId, event, mjolnir, parts);
-        } else if (parts[1] === 'directory' && parts.length > 3 && parts[2] === 'add') {
+        } else if (cmd === 'directory' && parts.length > 3 && parts[2] === 'add') {
             return await execAddRoomToDirectoryCommand(roomId, event, mjolnir, parts);
-        } else if (parts[1] === 'directory' && parts.length > 3 && parts[2] === 'remove') {
+        } else if (cmd === 'directory' && parts.length > 3 && parts[2] === 'remove') {
             return await execRemoveRoomFromDirectoryCommand(roomId, event, mjolnir, parts);
-        } else if (parts[1] === 'alias' && parts.length > 4 && parts[2] === 'add') {
+        } else if (cmd === 'alias' && parts.length > 4 && parts[2] === 'add') {
             return await execAddAliasCommand(roomId, event, mjolnir, parts);
-        } else if (parts[1] === 'alias' && parts.length > 3 && parts[2] === 'remove') {
+        } else if (cmd === 'alias' && parts.length > 3 && parts[2] === 'remove') {
             return await execRemoveAliasCommand(roomId, event, mjolnir, parts);
-        } else if (parts[1] === 'resolve' && parts.length > 2) {
+        } else if (cmd === 'resolve' && parts.length > 2) {
             return await execResolveCommand(roomId, event, mjolnir, parts);
-        } else if (parts[1] === 'powerlevel' && parts.length > 3) {
+        } else if (cmd === 'powerlevel' && parts.length > 3) {
             return await execSetPowerLevelCommand(roomId, event, mjolnir, parts);
-        } else if (parts[1] === 'shutdown' && parts[2] === 'room' && parts.length > 3) {
+        } else if (cmd === 'shutdown' && parts[2] === 'room' && parts.length > 3) {
             return await execShutdownRoomCommand(roomId, event, mjolnir, parts);
-        } else if (parts[1] === 'since') {
-            return await execSinceCommand(roomId, event, mjolnir, tokens);
-        } else if (parts[1] === 'kick' && parts.length > 2) {
+        } else if (cmd === 'since') {
+            return await execSinceCommand(roomId, event, mjolnir, lexer);
+        } else if (cmd === 'kick' && parts.length > 2) {
             return await execKickCommand(roomId, event, mjolnir, parts);
-        } else if (parts[1] === 'make' && parts[2] === 'admin' && parts.length > 3) {
+        } else if (cmd === 'make' && parts[2] === 'admin' && parts.length > 3) {
```

Comment:
Well, we should certainly figure out how to get rid of this giant switch statement and use a table to dispatch commands instead. We'd also be able to isolate commands and stop editing this file each time we need to add/remove one too by doing this.

Reply:
Absolutely, that's the objective of the refactoring.